airflow_tecton/operators/job_utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 Tecton, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import pprint
+import time
+
+
+def is_job_running(hook, workspace, feature_view, online, offline, allow_overwrite=False, start_time=None, end_time=None, job_type="batch"):
+    job = hook.find_materialization_job(
+        workspace=workspace,
+        feature_view=feature_view,
+        online=online,
+        offline=offline,
+        start_time=start_time,
+        end_time=end_time,
+        job_type=job_type,
+    )
+    if job:
+        logging.info(f"Existing job found: {pprint.pformat(job)}")
+        if job["state"].lower().endswith("running"):
+            logging.info(f"Job already in running state; not triggering new job")
+            return [job["id"]]
+        elif job["state"].lower().endswith("success"):
+            if allow_overwrite:
+                logging.info(
+                    "Overwriting existing job in success state; (allow_overwrite=True)"
+                )
+            else:
+                logging.info(
+                    f"Job already in success state; not triggering new job"
+                )
+                return [job["id"]]
+        else:
+            logging.info(f"Job in {job['state']} state; triggering new job")
+
+    return None
+
+
+def is_job_finished(hook, workspace, feature_view, online, offline, allow_overwrite=False, start_time=None, end_time=None, job_type="batch"):
```

Comment:
name here doesn't seem right. probably also needs comment, because it does a lot more than just check if the job is finished